app/frontend/src/components/ui/history-panel.tsx

```diff
@@ -28,27 +53,44 @@ export default function HistoryPanel({ show, history, onClosed, onSelectedGround
                 >
+                    <div className="sticky top-0 z-10 mb-4 flex items-center justify-between bg-white px-4 py-2">
+                        <h2 className="text-xl font-bold">{t("history.answerHistory")}</h2>
+                        <Button variant="ghost" size="sm" onClick={onClosed}>
+                            <X className="h-5 w-5" />
+                        </Button>
+                    </div>
                     <div className="p-4">
-                        <div className="mb-4 flex items-center justify-between">
-                            <h2 className="text-xl font-bold">{t("history.answerHistory")}</h2>
-                            <Button variant="ghost" size="sm" onClick={onClosed}>
-                                <X className="h-5 w-5" />
-                            </Button>
-                        </div>
                         {history.length > 0 ? (
-                            history.map((item, index) => (
-                                <div key={index} className="mb-6 border-b border-gray-200 pb-6">
-                                    <h3 className="mb-2 font-semibold">{item.id}</h3>
-                                    <pre className="mb-2 overflow-x-auto whitespace-pre-wrap rounded-md bg-gray-100 p-3 text-sm">
-                                        <code className="block h-24 overflow-y-auto">{item.transcript}</code>
-                                    </pre>
-                                    <div className="mt-2 flex flex-wrap gap-2">
-                                        {item.groundingFiles.map((file, index) => (
-                                            <GroundingFile key={index} value={file} onClick={() => onSelectedGroundingFile(file)} />
-                                        ))}
-                                    </div>
-                                </div>
-                            ))
+                            <div className="space-y-4">
+                                {history.map((item, index) => {
+                                    const nextItem = history[index + 1];
+                                    const isLast = index === history.length - 1;
+                                    const showTimestamp = shouldShowTimestamp(
+                                        new Date(item.timestamp),
+                                        nextItem ? new Date(nextItem.timestamp) : undefined,
+                                        isLast
+                                    );
+                                    return (
+                                        <div key={index}>
+                                            <div
+                                                className={`rounded-lg p-4 shadow ${item.sender === "user" ? "ml-auto bg-blue-100 pl-4" : "bg-gray-100"}`}
+                                                style={{ maxWidth: "75%" }} // Optional: Limit the width of the bubbles
```

Comment:
nit/suggestion: remove comment, doesn't seem to be adding value on top of the code